scripts/publish-to-npm.js

```diff
@@ -0,0 +1,96 @@
+// For example: 'release-retail-react-app-1.1.x'
+const RELEASE_ONE_PACKAGE = /release-([-a-z]+)-\d+\./i
+
+const main = () => {
+    // Exiting early if working tree is not clean
+    verifyCleanWorkingTree()
+
+    const branchName = sh.exec('git branch --show-current', {silent: true}).trim()
+    // DEBUG
+    // const branchName = 'release-3.0.x'
+    // const branchName = 'release-retail-react-app-1.0.x'
+
+    console.log('--- Given the current branch:', branchName)
+
+    const matched = branchName.match(RELEASE_ONE_PACKAGE)
+    const packageName = matched && matched[1]
+
+    if (packageName) {
+        console.log(`--- Releasing ${packageName}...`)
+        publishPackages([packageName])
+    } else {
+        console.log('--- Releasing all packages...')
+        publishPackages()
+    }
+}
+
+const publishPackages = (packages = []) => {
+    verifyCleanWorkingTree()
+
+    const publicPackages = JSON.parse(sh.exec('lerna list --json', {silent: true}))
+    const packagesToIgnore = publicPackages.filter((pkg) => !packages.includes(pkg.name))
+
+    const cleanUp = () => {
+        // Undo the temporary commit
+        sh.exec('git reset HEAD~1', {silent: true})
+
+        packagesToIgnore.forEach((pkg) => {
+            sh.exec('npm pkg delete private', {cwd: pkg.location})
+        })
+    }
+
+    const publishSomePackagesOnly = packages.length > 0
+    if (publishSomePackagesOnly) {
+        packagesToIgnore.forEach((pkg) => {
+            sh.exec('npm pkg set private=true', {cwd: pkg.location})
+        })
+
+        sh.exec('git add .', {silent: true})
+        sh.exec('git commit -m "temporary commit to have clean working tree"', {silent: true})
+    }
+
+    // Why do we still want `lerna publish`? It turns out that we do need it. Sometimes we wanted some behaviour that's unique to Lerna.
+    // For example: we have `publishConfig.directory` in some package.json files that only Lerna knows what to do with it.
+    // https://github.com/lerna/lerna/tree/main/libs/commands/publish#publishconfigdirectory
+
+    const {stderr, code} = sh.exec(
+        `npm run lerna -- publish from-package --yes --no-verify-access --pre-dist-tag next ${
+            process.env.CI ? '' : '--registry http://localhost:4873/'
+        }`
+    )
+    // DEBUG
+    // console.log('--- Would publish these public packages to npm:')
+    // sh.exec('lerna list --long')
+
+    // Make sure to clean up, no matter if there's an error or not
+    if (publishSomePackagesOnly) {
+        cleanUp()
+    }
+
+    if (stderr) {
+        process.exit(code)
+    }
+}
+
+const verifyCleanWorkingTree = () => {
+    const isWorkingTreeClean = sh.exec('git status --porcelain', {silent: true}).trim() === ''
+    if (!isWorkingTreeClean) {
+        console.error(
+            'There are some uncommitted changes. `lerna publish` expects a clean working tree.'
+        )
+        process.exit(1)
+    }
```

Comment:
I find this to be really annoying when testing generator... Is there a way to conditionally allowing publish with uncommitted changes?

Reply:
+1 on easing this restriction!

Reply:
`lerna publish` command does not allow you to go ahead with uncommitted changes. There is no such option to do that, unfortunately.

One workaround we can try next time is to programatically commit the changes and then undo the temporary commit afterwards.